content/en/blog/_posts/2025-04-17-volume-populators-ga/index.md

```diff
@@ -0,0 +1,56 @@
+---
+layout: blog
+title: "Kubernetes 1.33: Volume Populators Graduate to GA"
+date: 2025-04-17
+slug: volume-populators-ga
+author: >
+  Danna Wang (Google)
+  Sunny Song (Google)
+---
+
+The volume populators feature is now entering GA! The `AnyVolumeDataSource` feature
+gate defaults to enabled in Kubernetes v1.33, which means that users can specify any custom resource
+as the data source of a PVC.
+
+## What is new
+
+There are three major enhancements from beta.
+
+### Populator Pod is optional
+
+During the beta phase, we discovered that initiating data population doesn't always require a dedicated populator pod. In some cases, it's more efficient to interact with the cloud provider's APIs directly from the controller pod.
+
+To accommodate this, we've introduced three new plugin-based functions:
+* PopulateFn(): Executes the provider-specific data population logic.
+* PopulateCompleteFn(): Checks if the data population operation has finished successfully.
+* PopulateCleanupFn(): Cleans up temporary resources created by the provider-specific functions after data population is completed
+
+### Mutator functions to modify the Kubernetes resources
+
+For GA a MutatorConfig to the Volume Populator Controller is added, allowing the specification of mutator functions to modify Kubernetes resources. This enables customization of the default Kubernetes objects in the volume populator.
+
+### Flexible metric handling for providers
+
+Our beta phase highlighted a new requirement: the need to aggregate metrics not just from lib-volume-populator, but also from other components within the provider's codebase.
+
+To address this, we've introduced a provider metric manager. This enhancement delegates the implementation of metrics logic to the provider itself, rather than relying solely on lib-volume-populator. This shift provides greater flexibility and control over metrics collection and aggregation, enabling a more comprehensive view of provider performance.
```

Comment:
For an application to integrate with Volume Populators, what does it need to do to support this?

Reply:
There is no extra work you need to do, it is built in with lib-volume-populator.

Reply:
Can you add a hyper link to the provider metric manager?